File: lab3/lab3.py
import random
import math


def generate_password(alphabet, length):
    password = "".join(random.choice(alphabet) for _ in range(length))
    return password


def find_length(S, A):
    L = math.log(S, A)
    L = math.ceil(L)
    return L


def main(v: int, t: int, p_power: int):
    V = v * 24 * 60 * 7
    T = t / 7
    P = 10**-p_power

    # Выбираем алфавит
    alphabet = "0123456789abcdefghijklmnopqrstuvwxyz"

    # Вычисляем S
    A = len(alphabet)
    S = abs((V * T) / P)

    L = find_length(S, A)
    password = generate_password(alphabet, L)
    print("Сгенерированный пароль:", password)
    print("Длина:", L)


if __name__ == "__main__":
    V = int(input("Введите V: "))
    T = int(input("Введите T: "))
    P = int(input("Введите степень P: "))
    main(V, T, P)

File: lab3/lab3.php
<?php

function generate_password($alphabet, $length) {
    $password = "";
    $alphabetLength = strlen($alphabet);
    for ($i = 0; $i < $length; $i++) {
        $randomIndex = mt_rand(0, $alphabetLength - 1);
        $password .= $alphabet[$randomIndex];
    }
    return $password;
}

function find_length($S, $A) {
    $L = log($S, $A);
    $L = ceil($L);
    return $L;
}

function main($v, $t, $p_power) {
    $V = $v * 24 * 60 * 7;
    $T = $t / 7;
    $P = pow(10, -$p_power);

    // Выбираем алфавит
    $alphabet = "0123456789abcdefghijklmnopqrstuvwxyz";

    // Вычисляем S
    $A = strlen($alphabet);
    $S = abs(($V * $T) / $P);

    $L = find_length($S, $A);
    $password = generate_password($alphabet, $L);
    echo "Сгенерированный пароль: $password\n";
    echo "Длина: $L\n";
}

$V = (int) readline("Введите V: ");
$T = (int) readline("Введите T: ");
$P = (int) readline("Введите степень P: ");
main($V, $T, $P);

?>
